salt/transport/__init__.py

```diff
@@ -47,3 +51,25 @@ def factory(opts, **kwargs):
         # salt.transport.channel.Channel.factory()
         from salt.transport.client import ReqChannel
         return ReqChannel.factory(opts, **kwargs)
+
+
+class MessageClientPool(object):
+    def __init__(self, tgt, opts, args=None, kwargs=None):
+        sock_pool_size = opts['sock_pool_size'] if 'sock_pool_size' in opts else 1
+        if sock_pool_size < 1:
+            log.warn('sock_pool_size is not correctly set, \
+                     the option should be greater than 0 but, {0}'.format(sock_pool_size))
+            sock_pool_size = 1
+
+        if args is None:
+            args = ()
+        if kwargs is None:
+            kwargs = {}
+
+        self.message_clients = [tgt(*args, **kwargs) for _ in range(sock_pool_size)]
+
+    def __del__(self):
+        while self.message_clients:
+            message_client = self.message_clients.pop()
+            if message_client is not None:
+                del message_client
```

Comment:
Perhaps my understanding is incorrect. I would think that when this object is deleted, then `self.message_clients` is also deleted, which in turn would delete each individual message client. Is there a reason why this code is necessary?